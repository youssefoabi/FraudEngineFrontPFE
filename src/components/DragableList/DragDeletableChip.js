import { Draggable } from 'react-beautiful-dnd';
import { elementType, number } from 'prop-types';
import { isNilOrEmpty } from 'ramda-adjunct';
import Grid from '@material-ui/core/Grid';
import React from 'react';

const DragDeletableChip = ({ children, position, id }) => {
  if (isNilOrEmpty(children)) return null;

  const renderDraggableItem = provided =>
    provided && (
      <div ref={provided.innerRef} {...provided.draggableProps} {...provided.dragHandleProps}>
        {children}
      </div>
    );

  return (
    <Grid item>
      <Draggable key={id} draggableId={`${id}`} index={position}>
        {provided => renderDraggableItem(provided)}
      </Draggable>
    </Grid>
  );
};

DragDeletableChip.propTypes = {
  children: elementType,
  position: number,
  id: number,
};

DragDeletableChip.defaultProps = {
  children: null,
  position: null,
  id: null,
};

export default DragDeletableChip;
