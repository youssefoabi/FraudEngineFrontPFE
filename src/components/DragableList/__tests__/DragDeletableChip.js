import { itRendersAllMutations } from '@cdiscount/jest-utils';
import React from 'react';

import DragDeletableChip from '../DragDeletableChip';

jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  DragDropContext: ({ children }) => children,
}));

const mutations = [
  {
    name: 'without props',
    props: {},
  },
  {
    name: 'with props',
    props: { children: <div>item </div>, position: 2, id: 1 },
  },
];

describe(DragDeletableChip.name, () => {
  itRendersAllMutations(DragDeletableChip, mutations);
});
