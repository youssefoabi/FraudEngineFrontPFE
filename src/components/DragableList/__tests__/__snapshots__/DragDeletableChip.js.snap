// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragDeletableChip DragDeletableChip renders correctly without props 1`] = `""`;

exports[`DragDeletableChip DragDeletableChip renders correctly with props 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
>
  <Draggable
    draggableId="1"
    index={2}
    key="1"
  >
    <Component />
  </Draggable>
</WithStyles(ForwardRef(Grid))>
`;
